### ISSUE TITLE
imjournal read namespace specific journal data

### ISSUE THREAD

**Issue by nvoutsin** [rsyslog/rsyslog#4771]
Is there a way imjournal to read data coming from specific journal namespace (ie not the default namespace)?

**Comment by mpontillo** [comment]
From the `imjournal` documentation, I do not see a way to limit `imjournal`'s input to a single namespace. For what it's worth, I think this sounds like a useful enhancement. I want to use `imjournal` for a single service (which supports structured logging via `journald`), while continuing to use `imuxsock` for the remainder of the services on the system. Without the ability to scope the use of `imjournal` to a specific service, this would have the side effect of duplicating logs for other services (or risking sending them up in an unintended format, such as double JSON encoding, if they are already using `@cee` style logs).

**Comment by nvoutsin** [comment]
Indeed, the search for namespace support in imjournal, was triggered by the ability of systemd to define at service level a distinct namespace. The support of namespaces in imjournal would complement this feature pretty well and would clearly simplify the handling and routing of incoming messages.

**Comment by mpontillo** [comment]
For what it's worth (even though this is not ideal, because it implies a lot of unnecessary log processing) I was able to work around this as follows in order to experiment with `imjournal` for one particular application:
```
if $inputname == "imjournal" and $!_COMM == "<interesting-process>" then {
  call <my-ruleset>
  stop
}
if $inputname == "imjournal" then {
  stop
}
```
At first I tried to use `$programname` to filter the logs, but I found that it was always set to `journal`. Likewise, I modified the template in the documentation as follows in order to produce sane logging with accurate timestamps:
```
template(name="CEETemplate" type="string" string="%TIMESTAMP:::date-rfc3339% %HOSTNAME% %!_COMM%[%!_PID%]: @cee: %$!all-json%\n" )
```
